tls/extensions/s2n_server_renegotiation_info.c

```diff
@@ -102,6 +89,84 @@ static int s2n_renegotiation_info_recv(struct s2n_connection *conn, struct s2n_s
     return S2N_SUCCESS;
 }
 
+static int s2n_renegotiation_info_recv_renegotiation(struct s2n_connection *conn, struct s2n_stuffer *extension)
+{
+    POSIX_ENSURE_REF(conn);
+    uint8_t verify_data_len = conn->handshake.finished_len;
+    POSIX_ENSURE_GT(verify_data_len, 0);
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc5746#3.5
+     *# This text applies if the connection's "secure_renegotiation" flag is
+     *# set to TRUE (if it is set to FALSE, see Section 4.2).
+     */
+    POSIX_ENSURE(conn->secure_renegotiation, S2N_ERR_NO_RENEGOTIATION);
+
+    /**
+     *= https://tools.ietf.org/rfc/rfc5746#3.5
+     *# o  The client MUST then verify that the first half of the
+     *#    "renegotiated_connection" field is equal to the saved
+     *#    client_verify_data value, and the second half is equal to the
+     *#    saved server_verify_data value.  If they are not, the client MUST
+     *#    abort the handshake.
+     */
+
+    uint8_t renegotiated_connection_len = 0;
+    POSIX_GUARD(s2n_stuffer_read_uint8(extension, &renegotiated_connection_len));
+    POSIX_ENSURE(verify_data_len * 2 == renegotiated_connection_len, S2N_ERR_NO_RENEGOTIATION);
```

Comment:
Yeah I was debating just using "S2N_ERR_BAD_MESSAGE" like we usually do, but I figured "S2N_ERR_NO_RENEGOTIATION" might be more useful since something went wrong with the "secure" part of "secure renegotiation"